src/chainstate/stacks/index/storage.rs

```diff
@@ -0,0 +1,1213 @@
+
+    /// Get the block hash of the "parent of the root".  This does not correspond to a real block,
+    /// but instead is a sentinel value that is all 1's
+    pub fn block_sentinel() -> BlockHeaderHash {
+        BlockHeaderHash([255u8; BLOCK_HEADER_HASH_ENCODED_SIZE as usize])
+    }
+
+    #[cfg(test)]
+    pub fn stats(&mut self) -> (u64, u64) {
+        let r = self.read_count;
+        let w = self.write_count;
+        self.read_count = 0;
+        self.write_count = 0;
+        (r, w)
+    }
+    
+    #[cfg(test)]
+    pub fn node_stats(&mut self) -> (u64, u64, u64) {
+        let nr = self.read_node_count;
+        let br = self.read_backptr_count;
+        let nw = self.write_node_count;
+
+        self.read_node_count = 0;
+        self.read_backptr_count = 0;
+        self.write_node_count = 0;
+
+        (nr, br, nw)
+    }
+
+    #[cfg(test)]
+    pub fn leaf_stats(&mut self) -> (u64, u64) {
+        let lr = self.read_leaf_count;
+        let lw = self.write_leaf_count;
+
+        self.read_leaf_count = 0;
+        self.write_leaf_count = 0;
+
+        (lr, lw)
+    }
+
+    // last two bytes form the directory name
+    pub fn block_dir(dir_path: &String, bhh: &BlockHeaderHash) -> PathBuf {
+        let bhh_bytes = bhh.as_bytes();
+        let bhh_1 = format!("{:02x}", bhh_bytes[31]);
+        let bhh_2 = format!("{:02x}", bhh_bytes[30]);
+        let p = Path::new(dir_path)
+                    .join(bhh_1)
+                    .join(bhh_2);
+        p
+    }
+
+    pub fn block_path(dir_path: &String, bhh: &BlockHeaderHash) -> PathBuf {
+        // it looks awkward, but it's waaaay faster than just doing to_hex()
+        let bhh_bytes = bhh.as_bytes();
+        let bhh_name = format!("{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}",
+                              bhh_bytes[0],     bhh_bytes[1],       bhh_bytes[2],       bhh_bytes[3],
+                              bhh_bytes[4],     bhh_bytes[5],       bhh_bytes[6],       bhh_bytes[7],
+                              bhh_bytes[8],     bhh_bytes[9],       bhh_bytes[10],      bhh_bytes[11],
+                              bhh_bytes[12],    bhh_bytes[13],      bhh_bytes[14],      bhh_bytes[15],
+                              bhh_bytes[16],    bhh_bytes[17],      bhh_bytes[18],      bhh_bytes[19],
+                              bhh_bytes[20],    bhh_bytes[21],      bhh_bytes[22],      bhh_bytes[23],
+                              bhh_bytes[24],    bhh_bytes[25],      bhh_bytes[26],      bhh_bytes[27],
+                              bhh_bytes[28],    bhh_bytes[29],      bhh_bytes[30],      bhh_bytes[31]);
+
+        TrieFileStorage::block_dir(dir_path, bhh).join(bhh_name)
+    }
+
+    pub fn block_path_tmp(dir_path: &String, bhh: &BlockHeaderHash) -> PathBuf {
+        // it looks awkward, but it's waaaay faster than just doing to_hex()
```

Comment:
did you compare these in `--release` mode? I think we should try to avoid optimizations like these whenever possible.

Reply:
No, but I will do so.  Will remove this if perf shows no difference between the two.